ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -0,0 +1,394 @@
+| ACP | 194 |
+| :--- | :--- |
+| **Title** | Streaming Asynchronous Execution |
+| **Author(s)** | Arran Schlosberg ([@ARR4N](https://github.com/ARR4N)), Stephen Buttolph ([@StephenButtolph](https://github.com/StephenButtolph)) |
+| **Status** | Proposed |
+| **Track** | Standards |
+
+
+## Abstract
+
+Streaming Asynchronous Execution (SAE) decouples consensus and execution by introducing a (theoretically unbounded) transaction queue upon which consensus is performed.
+A concurrent execution stream is responsible for clearing the queue and reporting a delayed state root for recording by later rounds of consensus.
+Validation of transactions to be pushed to the queue is lightweight but guarantees eventual execution.
+
+## Motivation
+
+### Performance improvements
+
+1. Concurrent consensus and execution streams eliminate node context switching, reducing latency caused by each waiting on the other.
+In particular, "VM time" (akin to CPU time) more closely aligns with wall time since it is no longer eroded by consensus.
+This increases gas per wall-second even without an increase in gas per VM-second.
+2.  Lean, execution-only clients can rapidly execute the queue agreed upon by consensus, providing accelerated receipt issuance and state computation.
+Without the need to compute state _roots_, such clients can eschew expensive Merkle data structures.
+End users see expedited but identical transaction results.
+3. Irregular stop-the-world events like database compaction are amortised over multiple blocks.
+4. EOA-to-EOA transfers of value can be considered settled immediately upon being pushed to the queue by consensus.
+5. Introduces bursty throughput under temporary high load, preemptively accepting transactions that may take marginally longer to execute, without a reduction in security guarantees.
+
+### User stories
+
+1. A sophisticated DeFi trader runs a highly optimised execution client, locally clearing the transaction queue well in advance of the network—setting the stage for HFT DeFi.
+2. A custodial platform filters the queue for only those transactions sent to one of their EOAs, immediately crediting user balances.
+
+## Specification
+
+### Transaction queue
+
+#### Pushing transactions
+
+Standard, synchronous execution couples a block to the ordered set of transactions to be executed in said block.
+Instead, let there be a FIFO queue of all known, valid transactions—the specifics of "validity" will be addressed later.
+Under SAE, a block's ordered transactions instead constitute those to be pushed to the queue
+A transaction is said to be _included_ when a block builder adds it to a block and _accepted_ once consensus, performed over blocks, agrees to add it to the queue.
```

Comment:
> A transaction is said to be _included_ when a block builder adds it to a block and _accepted_ once consensus, performed over blocks, agrees to add it to the queue.

This has the added benefit that we can incorporate verifiable randomness seamlessly, since a transaction is only added to the queue once consensus totally orders it, and by then avalanchego will have its randomness. 

In the current way Avalanchego works, the transactions are executed before consensus, so the randomness is not yet available.

Reply:
What's the VRF in such a scenario?